lib/client-options.js

```diff
@@ -89,6 +90,9 @@ function extend(baseOptions, userOptions) {
   if (!(options.policies.addressResolution instanceof policies.addressResolution.AddressTranslator)) {
     throw new TypeError('Address resolution policy must be an instance of AddressTranslator');
   }
+  if (!(options.policies.timestampGeneration instanceof policies.timestampGeneration.TimestampGenerator)) {
```

Comment:
It would be good if there was a way to disable timestamp generators (from what I can tell this prevents this) as there are reasons (although not as many) to not use client timestamps, i.e. if you have a large number of clients updating the same rows and you can't reliably ensure they will be time synched.